docs/introduction.md

```diff
@@ -4,48 +4,104 @@ title: Introduction
 ---
 
-Unmock provides an opinionated, easy way to create and manage mocks for external APIs.
+unmock is an opinionated and scalable way to mock external services in tests.
 
 ## Motivation
 
+Services are the glue connecting modern applications: for example, the GitHub API is a service. Every service has a specification describing how the service behaves, be either written documentation, [OpenAPI](https://www.openapis.org/), [RAML](https://raml.org/), or something else. The specification for a service is _reusable_ across applications.
+
+Still, when testing how our applications integrate with external services, we rarely use the specifications directly. Instead, we write adhoc rules for how the services should behave. Writing such rules is error-prone, time-consuming, and hard to maintain.
+
+This is what unmock wants to fix. Testing the integration with external services should start from the _service specification_. Setting the service _state_ should happen programmatically before every test. The state should be _consistent_ with the service specification.
+
 ## How it works
 
-Let's assume you want to fetch a GitHub user's email address using `fetch`:
+Let's assume you want to fetch a GitHub organization's email address using `fetch`:
 
 ```js
-async function fetchGitHubUserEmail(username) {
+async function fetchGitHubOrganizationEmail(organization) {
   const headers = new Headers({
     Authorization: `Bearer ${process.ENV.GITHUB_API_TOKEN}`,
   });
-  const userData = await fetch(
-    `https://api.github.com/users/${username}`,
+  const organizationData = await fetch(
+    `https://api.github.com/orgs/${organization}`,
     headers
   );
-  return userData.email;
+  return organizationData.email;
+}
+```
+
+Your test looks like this:
+
+```js
+it("fetches GitHub organization's email", async () => {
+  const email = await fetchGitHubOrganizationEmail("meeshkan");
+  expect(email).toBe("contact@meeshkan.com");  // email address for Meeshkan organization
 }
 ```
 
-You want to test the logic in your function and write a test looking like this:
+The test would complete successfully if the `GITHUB_API_TOKEN` environment variable was set properly. However, it is bad practice to hit the API in unit tests, because
+
+1. running unit tests counts against the rate limit,
+1. bugs in the code could corrupt production data,
+1. organization's email could change and that would break the test.
+
+Let us add unmock to capture outgoing calls and check if the value returned from `fetchGitHubOrganizationEmail` is a `String`:
 
 ```js
-it("fetches GitHub user email", async () => {
-  const email = await fetchGitHubUserEmail("meeshkan");
-  expect(email).toBeDefined();
+import unmock from "unmock-node";
+
+beforeAll(() => {
+  unmock.initialize();  // Capture outgoing calls
+})
+it("fetches GitHub organization's email", async () => {
+  const email = await fetchGitHubOrganizationEmail("meeshkan");
+  expect(email).toEqual(expect.any(String));
```

Comment:
Why not `expect(typeof email).toEqual("string")`?

Reply:
Why not `expect(email).toEqual(expect.any(String));` 🙂

Reply:
Changed to `typeof` check.